crates/sqllogictest/src/engine/mod.rs

```diff
@@ -17,29 +17,45 @@
 
 mod datafusion;
 
+use std::collections::HashMap;
 use std::path::Path;
 
 use anyhow::anyhow;
+use serde::Deserialize;
 use sqllogictest::{AsyncDB, MakeConnection, Runner, parse_file};
-use toml::Table as TomlTable;
 
 use crate::engine::datafusion::DataFusionEngine;
 use crate::error::{Error, Result};
 
-const TYPE_DATAFUSION: &str = "datafusion";
+/// Configuration for the catalog used by an engine
+#[derive(Debug, Clone, Deserialize)]
+pub struct CatalogConfig {
```

Comment:
```suggestion
pub struct DatafusionCatalogConfig {
```

We will support more engines in future, for example spark engine, and this catalog config is datafusion specific.